envoy/api/v2/core/http_uri.proto

```diff
@@ -0,0 +1,59 @@
+
+syntax = "proto3";
+
+package envoy.api.v2.core;
+
+import "validate/validate.proto";
+
+// When Envoy makes a HTTP call, two values are needed to specify the HTTP
+// target; "uri" for the HTTP URI and "cluster" for the cluster name in the
+// Envoy config.
+// [#not-implemented-hide:]
+message HttpURI {
+  // The HTTP server URI. It should be a full FQDN with protocol, host and path.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //    uri: https://www.googleapis.com/oauth2/v1/certs
+  //
+  string uri = 1 [(validate.rules).string.min_bytes = 1];
+
+  // Specify the cluster. Only static cluster is supported for now.
+  oneof cluster {
```

Comment:
nit: I might call this oneof "endpoint" or something else, since in the future it might not be an Envoy cluster at all but a true URI that can be called directly?